### DISCUSSION TITLE
Problems with koboldcpp

### DISCUSSION THREAD

**Discussion by Alexey688** [LostRuins/koboldcpp#82] [Q&A]
After opening koboldcpp (I tried different versions), I can't specify the path to the model, no characters are entered and the window closes after 15-20 seconds. I'm not good at all this, please explain in simple language.

**Comment by gustrd** [comment]
Try opening it at console (command prompt, cmd). Post here the message that is being sent.

**Reply by Alexey688** [reply]
![2023-04-17_22-46-15](https://user-images.githubusercontent.com/131031506/232594395-27e0df13-b77a-4e36-99ec-3505eebeb9b2.png)
![2023-04-17_22-45-33](https://user-images.githubusercontent.com/131031506/232594410-c428a22b-8600-48fb-a9d2-a852c6480305.png)
I tried it before, all the same

**Reply by Alexey688** [reply]
![2023-04-17_22-49-38](https://user-images.githubusercontent.com/131031506/232595097-e929f768-7c30-4632-8907-cf8d9d69d571.png)
I also open the model using koboldcpp, but still the window closes after a while.

**Comment by gustrd** [comment]
Do you have enough memory at your computer?

**Reply by Alexey688** [reply]
![2023-04-17_22-53-19](https://user-images.githubusercontent.com/131031506/232595794-1b3565e6-d264-44cc-ad63-e2080cfa0b4b.png)
yes

**Reply by Alexey688** [reply]
![2023-04-17_23-00-41](https://user-images.githubusercontent.com/131031506/232597733-b9fb7c89-ee54-488f-a606-ff8c71a9ac01.png)
all the same

**Comment by gustrd** [comment]
Sorry, I mean RAM Memory. How much free RAM do you have before loading? It need at least the .bin file size, a little more.

**Reply by Alexey688** [reply]
![2023-04-17_23-28-35](https://user-images.githubusercontent.com/131031506/232602904-de7b146b-9f02-4674-b3a7-bba6d0ea6403.png)
I have 12 GB of RAM, that is, even with the browser running, 60% ~7-8 GB remain free.
In the screenshot, I managed to launch something, is this not how it should be?

**Reply by gustrd** [reply]
@LostRuins , I find strange that here he could start loading a model.

**Reply by LostRuins** [reply]
@gustrd His device probably does not support avx2, and he has not used the --noavx2 flag.

**Reply by gustrd** [reply]
@LostRuins the Cyrillic characters at his path could be an issue also?

**Reply by gustrd** [reply]
The Cyrillic is not the issue, I tried to reproduce the error but it worked ok. Probably it's AVX2.

**Comment by gustrd** [comment]
I think that may be the problem, try with a smaller model first, like ggml-alpaca-7b-q4 ( ex: https://huggingface.co/Sosaka/Alpaca-native-4bit-ggml ).

**Reply by Alexey688** [reply]
So I have a problem because before choosing a model, koboldcpp closes and I can't enter anything

**Reply by Alexey688** [reply]
This problem is before the model selection

**Reply by gustrd** [reply]
Your printscreen shows a model being loaded. I don't understand. Have you tried just clicking the exe, not the shortcut?

**Reply by Alexey688** [reply]
I've already tried opening it a lot

**Reply by gustrd** [reply]
So I have no other ideas, sorry.

**Comment by LostRuins** [comment]
@Alexey688 your device may not support the AVX2 instruction. 

Can you try opening it with from the `Run` window with the extra flag `--noavx2`?

**Reply by Alexey688** [reply]
![2023-04-18_13-30-14](https://user-images.githubusercontent.com/131031506/232750942-9f4bf90d-0ea3-4354-b56d-cb5c2d27919e.png)
Maybe the problem is a weak processor and video memory? koboldcpp just closes or crashes when I run the model

**Reply by LostRuins** [reply]
![image](https://user-images.githubusercontent.com/39025047/232751489-5a80191d-dc4d-415b-b444-81e74ffea545.png)
Please try running with `--noavx2` flag

**Reply by Alexey688** [reply]
Immediately closes after the lines appear

**Reply by LostRuins** [reply]
Please send a screenshot similar to the above, showing the output after using the `--noavx2` flag

**Reply by Alexey688** [reply]
![2023-04-18_14-05-50](https://user-images.githubusercontent.com/131031506/232758752-a113fd0f-7832-4bd0-95ef-b9cfdfa6dbaa.png)
![2023-04-18_14-06-17](https://user-images.githubusercontent.com/131031506/232758760-54fd74a6-7c5c-4d4a-8fe5-5b900a375004.png)
after a second, the window closes

**Reply by LostRuins** [reply]
Okay, does the choose file window appear?

**Reply by Alexey688** [reply]
No, nothing appears there and I can't write anything

**Reply by LostRuins** [reply]
Okay, try run this command which avoids using the GUI. You need to manually pass in the model file path when running.

run `C:\koboldcpp.exe --noavx2 --model C:\modelfoldername\ggml-model-q4_0.bin`

Replace with the location of your model.bin file

**Reply by Alexey688** [reply]
The window also closes quickly when the model is defined

**Reply by LostRuins** [reply]
In that case, maybe your PC is not compatible with the even AVX1 instructions, and KoboldCpp will not work. 

I'll leave this issue open for a while, in case anyone else has the same problem or found a solution...
